plugins/terraform/layer0/data_source_layer0_service.go

```diff
@@ -0,0 +1,69 @@
+package layer0
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/quintilesims/layer0/client"
+)
+
+func dataSourcelayer0Service() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceLayer0ServiceRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"environment_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"environment_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"scale": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceLayer0ServiceRead(d *schema.ResourceData, meta interface{}) error {
+	apiClient := meta.(client.Client)
+
+	query := url.Values{}
+	query.Set(client.TagQueryParamType, "service")
+	query.Set(client.TagQueryParamName, d.Get("name").(string))
+	query.Set(client.TagQueryParamEnvironmentID, d.Get("environment_id").(string))
+
+	tags, err := apiClient.ListTags(query)
+	if err != nil {
+		return err
+	}
+
+	if len(tags) == 0 {
+		return fmt.Errorf("No service found matching %#v", query)
+	}
+
+	serviceID := tags[0].EntityID
+	service, err := apiClient.ReadService(serviceID)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(service.ServiceID)
```

Comment:
[nit] Is it better to compare this too a struct? I looked into it but it seems to complicate it more. Just wanted an opinion.

Reply:
I'm not sure what you mean - what's being compared here?